src/NFun.Fuspec.TestHandler/FuspecTestCasesRunner.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Diagnostics;
+using Nfun.Fuspec.Parser.Model;
+using NFun.Fuspec.TestHandler.Models;
+using NFun.ParseErrors;
+using NFun.Runtime;
+
+namespace NFun.Fuspec.TestHandler
+{
+    public static class FuspecTestCasesRunner
+    {
+        public static BaseFuspecTestCaseResult RunFuspecTest(FuspecTestCase testCase)
+        {
+            var stopwatch = Stopwatch.StartNew();
+            return ParseTestResult(testCase, stopwatch);
+        }
+
+        private static BaseFuspecTestCaseResult ParseTestResult(FuspecTestCase testCase, Stopwatch stopwatch)
+        {
+            BaseFuspecTestCaseResult testResult;
+
+            try
+            {
+                //todo parse in and out funResult
+                FunRuntime funResult = FunBuilder.With(testCase.Script).Build();
+                testResult = TestResultConverter.GoodTestResult(stopwatch.ElapsedMilliseconds, testCase);
+            }
+            catch (FunParseException parseException)
+            {
+                testResult = TestResultConverter.BadTestResult(stopwatch.ElapsedMilliseconds, testCase, parseException);
```

Comment:
You need two types of "badTestResult":
- ParseError (when you catch funParseException) with whole fun parse exception information  
- NfunFailure (when you catch all other exception at stage of Build() ) with concrete exception exemplar for further analysis

Reply:
Why do I need two types of errors? The given class has a property with an error, I save the main error, and then it is possible to determine by type what the error was

Reply:
So you make weak typing here to return strict typing in the calling code ;) Why?

You got 3 cases now. It is great if you show it by types. Like enums